azure-toolkit-libs/azure-toolkit-common-lib/src/main/java/com/microsoft/azure/toolkit/lib/common/model/AbstractAzService.java

```diff
@@ -106,7 +107,32 @@ protected <E> E doGetById(@Nonnull String id) {
             resourceId = resourceId.parent();
         }
         for (Pair<String, String> resourceTypeName : resourceTypeNames) {
-            resource = resource.getSubModule(resourceTypeName.getLeft()).getOrDraft(resourceTypeName.getRight(), resourceGroup);
+            resource = Optional.ofNullable(resource)
+                .map(r -> r.getSubModule(resourceTypeName.getLeft()))
+                .map(m -> m.getOrDraft(resourceTypeName.getRight(), resourceGroup)).orElse(null);
+        }
+        return (E) resource;
+    }
+
+    @Nullable
+    public <E> E getOrInitById(@Nonnull String id) { // move to upper class
+        return this.doGetOrInitById(id);
+    }
+
+    @Nullable
+    protected <E> E doGetOrInitById(@Nonnull String id) {
+        ResourceId resourceId = ResourceId.fromString(id);
+        final String resourceGroup = resourceId.resourceGroupName();
+        AbstractAzResource<?, ?, ?> resource = Objects.requireNonNull(this.get(resourceId.subscriptionId(), resourceGroup));
+        final LinkedList<Pair<String, String>> resourceTypeNames = new LinkedList<>();
+        while (resourceId != null) {
+            resourceTypeNames.push(Pair.of(resourceId.resourceType(), resourceId.name()));
+            resourceId = resourceId.parent();
+        }
+        for (Pair<String, String> resourceTypeName : resourceTypeNames) {
```

Comment:
May need to handle case that module is not supported, which will return null rather than a `GenericResource`